Fixes plasmamen dying from liver damage when consuming plasma & other organic toxins, fixes plasma fixation for plasmamen and other species

vinylspiders opened the pull request:
## About The Pull Request

(edited for brevity---AS BEST I COULD, this turned into a lot more fixes than I set out to do)

Instead of actually removing and metabolizing plasma in the liver, plasmamen were having plasma removed in a special handle_chemicals proc.  

The issue with this is that the liver's on_life proc would still fire off alongside it, detecting reagents in the mob (because handle_chemicals removes them little by little, at the same rate as the liver does). And because the liver code sees it as a toxin, it takes damage until it fails, leading to death as described in #72229.

Not actually metabolizing plasma also led to another issue, as mentioned in #61424. 

The solution after much faffing around ended up being to move the actual reagent removal from handle_chemicals to the liver code, and to compare a toxin's affected_organtype to the liver's 'status' var to check if the toxin can damage the liver. There was currently only an ORGAN_ORGANIC and ORGAN_ROBOTIC defined for this, so I added ORGAN_MINERAL for plasma livers.

It still works as before, but now the reagent will actually get all the metabolism procs called on it and be removed by the liver. And mobs with mineral livers (currently only plasmamen) do not suffer organ failure from processing organic toxins.

Plasma fixation itself also needed some TLC, as it was likely written before liver damage was a thing. A lot of this comes down to rogue livers operating independently of their species now that I think about it...

Fixed it so you no longer take liver damage from injected/consumed plasma when under the effects of the stage 4+ virus by making use of the new PLASMA_LOVER_METABOLISM liver trait.

---

On top of that made it so that breathing in plasma through internals also contributes to plasma fixation healing. This is a slight buff to virology (especially for plasmamen) but it didn't seem busted or anything from my testing. Can lead to some interesting subtle approaches to healing for virologists as they can use internals rather than just flooding rooms with plasma...

Some notes: 

- due to the way lungs are coded, non-plasma breathing species take a significant amount of tox damage from breathing plasma through internals in most cases and still need the right gas to be present so as to not asphyxiate. 
- During my testing this damage did not get healed fast enough to make it viable for humans to wear plasma internals as found in spawned tanks and such.
- I did not play around with creating custom internals mixes of O2 and plasma but in theory that could possibly allow humans to heal faster than the tox damage is applied. I think the tox damage on breath is based on the amount of mols in the tank so perhaps a small amount of plasma in the mix could work. 
- As it stands the heal is very slight at 4kpa which is the typical tank pressure for plasma internals. Going up to 15kpa can result in a stronger heal (on par with injecting plasma) but going further offers no additional benefit, and internals will be consumed much more quickly.
- This heal bonus is multiplicative with the rest, but from my testing it still stays within reasonable numbers even with all 3 sources going at once so as to not feel broken.
- Plasma fixation does not heal organ damage, but if I could add that in easily enough should it be wanted.

The numbers themselves can be easily tweaked if desired, should people find it too strong or what have you!

## Why It's Good For The Game

Fixes #72229
Fixes #61424
Fixes #70460 --not sure if this PR fixed it but I was able to infect monkeys and plasmamen with a virus containing Inorganic Biology in my testing, so this may be closed

Plasmamen dying from plasma is a travesty and reparations must be made.


## Changelog

:cl:
fix: plasmamen no longer can suffer liver failure from injecting themselves with plasma (unless they have a human liver for some reason).
fix: a person who is infected with a plasma fixation disease can no longer can suffer liver failure from injecting themselves with plasma.
fix: plasmamen no longer can suffer liver failure from consuming other organic toxins, which they are not supposed to be affected by. 
fix: plasma fixation heal-through-inject-or-consumption now works for plasmamen by letting the metabolism procs take care of removing it from their systems.
fix: hot ice now has the same wound-healing and nontoxic properties as plasma to plasmamen, since it is described as "frozen plasma". It is also nontoxic to plasma fixation virus sufferers.
fix: plasma breathed through internals now contributes to the healing amount from plasma fixation.
/:cl:

vinylspiders committed:
De-snowflaking of plasmamen handling of plasma which allows for actual metabolism of it

Also fixes liver damage still being applied because livers have their own proc which was getting called concurrently alongside handle_chemicals

timothymtorres commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -106,6 +106,8 @@
 	if(holder.has_reagent(/datum/reagent/medicine/epinephrine))
 		holder.remove_reagent(/datum/reagent/medicine/epinephrine, 2 * REM * delta_time)
 	affected_mob.adjustPlasma(20 * REM * delta_time)
+	if(toxpwr && isplasmaman(affected_mob)) // plasma is not toxic to them
```
You should add a `TRAIT_PLASMA_LOVER_METABOLISM` trait for plasmaman livers and move all the effects to it.  See how it's done for security liver:

https://github.com/tgstation/tgstation/blob/46594206f04eac960d6eb5baacc980e10df6e949/code/modules/reagents/chemistry/reagents/alcohol_reagents.dm#L1006-L1014

timothymtorres commented on code/modules/surgery/organs/liver.dm:
```diff
@@ -107,16 +107,22 @@
-			if(provide_pain_message != HAS_PAINFUL_TOXIN)
-				provide_pain_message = toxin.silent_toxin ? HAS_SILENT_TOXIN : HAS_PAINFUL_TOXIN
-
+			// Plasmamen don't take liver damage from plasma, but instead heal wounds
+			if(isplasmaman(liver_owner) && istype(toxin, /datum/reagent/toxin/plasma))
+				for(var/i in owner.all_wounds)
+					var/datum/wound/iter_wound = i
+					iter_wound.on_xadone(4 * REAGENTS_EFFECT_MULTIPLIER * delta_time) // plasmamen use plasma to reform their bones or whatever
+			else
+				var/amount = round(toxin.volume, CHEMICAL_QUANTISATION_LEVEL) // this is an optimization
+				if(belly)
+					amount += belly.reagents.get_reagent_amount(toxin.type)
+
+				// a 15u syringe is a nice baseline to scale lethality by
+				liver_damage += ((amount/15) * toxin.toxpwr) / liver_resistance
+
+				if(provide_pain_message != HAS_PAINFUL_TOXIN)
+					provide_pain_message = toxin.silent_toxin ? HAS_SILENT_TOXIN : HAS_PAINFUL_TOXIN
+	
```
Rewrite this so that you use plasma lover liver trait instead of `isplasmaman`.  

It's also better to use a `continue` statement with the if loop so you don't have to nest your code deeper than it needs to be.

```
for(var/datum/reagent/toxin/toxin in cached_reagents)
	// Plasmamen don't take liver damage from plasma, but instead heal wounds
	if(isplasmaman(liver_owner) && istype(toxin, /datum/reagent/toxin/plasma))
		for(var/i in owner.all_wounds)
			var/datum/wound/iter_wound = i
			iter_wound.on_xadone(4 * REAGENTS_EFFECT_MULTIPLIER * delta_time) // plasmamen use plasma to reform their bones or whatever
		continue

	var/amount = round(toxin.volume, CHEMICAL_QUANTISATION_LEVEL) // this is an optimization
	if(belly)
		amount += belly.reagents.get_reagent_amount(toxin.type)

	// a 15u syringe is a nice baseline to scale lethality by
	liver_damage += ((amount/15) * toxin.toxpwr) / liver_resistance

	if(provide_pain_message != HAS_PAINFUL_TOXIN)
		provide_pain_message = toxin.silent_toxin ? HAS_SILENT_TOXIN : HAS_PAINFUL_TOXIN
```

vinylspiders replied:
Gotcha -- good suggestions! I was not able to find a 'plasma lover liver trait' however, so I checked for the type of liver instead... If that's not an elegant enough solution I can change it, let me know if I missed something there.

There is one issue, and I'm going to put it back to draft for now about it---although this certainly fixes #72644 I am testing for #61424 and it seems that there is another hidden issue somewhere deeper in the code, unrelated to what I fixed.

The plasmaman I was trying to inject with a virus for testing was simply not getting the virus, when I checked its 'diseases' list in vars. Whereas the monkey I was testing on got it right away. I will have to look into this later today but if you have any ideas let me know.

vinylspiders replied:
A followup to this, I found the issue! Everything seems to be working now. Also see #72609 for another plasmaman fix.

timothymtorres commented:
Very nice!  Been wanting to do this myself at some point in the future but I have way to much stuff already on my TODO list.

vinylspiders committed:
Update - checks for plasmaman liver instead of plasmaman race

Since the liver is handling it. Can perhaps allow for organ transplant shenanigans

vinylspiders committed:
Update liver.dm

vinylspiders committed:
Switch to using traits, add a new trait for plasma metabolizing livers

vinylspiders committed:
Whoops

vinylspiders committed:
Fixes issue that was causing plasmamen to be immune to disease

private-tristan commented:
> Fixes another issue where plasmamen were immune to ALL disease apart from parrot possession due to a their lack of MOB_ORGANIC flag.

isn't this intended?
https://github.com/tgstation/tgstation/blob/27f09dd015ed86195001e3170419be2878563902/code/modules/mob/living/carbon/human/species_types/plasmamen.dm#L207-L210

vinylspiders committed:
Merge branch 'master' into plasmaman-tox-fix

vinylspiders committed:
Moved plasmaman bone healing into on_mob_life

Moves plasmamen code out of liver so the istype call is unneeded.

Removed the virus flag, turns out there is already a way of making a disease target plasmamen

vinylspiders commented:
> > Fixes another issue where plasmamen were immune to ALL disease apart from parrot possession due to a their lack of MOB_ORGANIC flag.
> 
> isn't this intended?
> 
> https://github.com/tgstation/tgstation/blob/27f09dd015ed86195001e3170419be2878563902/code/modules/mob/living/carbon/human/species_types/plasmamen.dm#L207-L210

You're right, it's been years since I have played viro and had no idea there was another step needed to make a virus target plasmamen.

vinylspiders commented on code/modules/surgery/organs/liver.dm:
```diff
@@ -107,6 +107,13 @@
+			// Plasmamen don't take liver damage from plasma, but instead heal wounds
+			if(HAS_TRAIT(src, TRAIT_PLASMA_LOVER_METABOLISM) && istype(toxin, /datum/reagent/toxin/plasma))
+				for(var/i in owner.all_wounds)
+					var/datum/wound/iter_wound = i
```
Done

vinylspiders commented on code/modules/surgery/organs/liver.dm:
```diff
@@ -107,6 +107,13 @@
+					var/datum/wound/iter_wound = i
+					iter_wound.on_xadone(4 * REAGENTS_EFFECT_MULTIPLIER * delta_time) // plasmamen use plasma to reform their bones or whatever
+				continue
+
```
You're right--I have moved that to on_mob_life() for the plasma reagent.

timothymtorres commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -106,6 +106,12 @@
 	if(holder.has_reagent(/datum/reagent/medicine/epinephrine))
 		holder.remove_reagent(/datum/reagent/medicine/epinephrine, 2 * REM * delta_time)
 	affected_mob.adjustPlasma(20 * REM * delta_time)
+	var/obj/item/organ/internal/liver/liver = affected_mob.getorganslot(ORGAN_SLOT_LIVER)
+	if(liver && HAS_TRAIT(liver, TRAIT_PLASMA_LOVER_METABOLISM)) // Some livers can metabolize plasma
+		toxpwr = 0
+		if(isplasmaman(affected_mob)) // having wounds repaired still requires plasmaman biology, in case of transplanted plasma livers
+			for(var/datum/wound/iter_wound as anything in affected_mob.all_wounds)
+				iter_wound.on_xadone(4 * REAGENTS_EFFECT_MULTIPLIER * delta_time) // plasmamen use plasma to reform their bones or whatever
```
```
/datum/reagent/toxin/plasma/on_mob_metabolize(mob/living/carbon/affected_mob)
	var/obj/item/organ/internal/liver/liver = affected_mob.getorganslot(ORGAN_SLOT_LIVER)
	if(liver && HAS_TRAIT(liver, TRAIT_PLASMA_LOVER_METABOLISM)) // Some livers can metabolize plasma
		toxpwr = 0

/datum/reagent/toxin/plasma/on_mob_end_metabolize(mob/living/carbon/affected_mob)
	toxpwr = initial(toxpwr)

/datum/reagent/toxin/plasma/on_mob_life(mob/living/carbon/affected_mob, delta_time, times_fired)
	if(holder.has_reagent(/datum/reagent/medicine/epinephrine))
		holder.remove_reagent(/datum/reagent/medicine/epinephrine, 2 * REM * delta_time)
	affected_mob.adjustPlasma(20 * REM * delta_time)
	if(isplasmaman(affected_mob)) // having wounds repaired still requires plasmaman biology, in case of transplanted plasma livers
		for(var/datum/wound/iter_wound as anything in affected_mob.all_wounds)
			iter_wound.on_xadone(4 * REAGENTS_EFFECT_MULTIPLIER * delta_time) // plasmamen use plasma to reform their bones or whatever
	return ..()
```

CC @MrMelbert how does this look?  

- Plasmaman liver determines if they take toxin damage from plasma
- Plasmaman biology heals wounds separately (even without the liver)

So a human with a plasmaman liver, can ignore plasma toxin but without the regen abilities.  A plasmaman with a human liver will take toxin damage, but the rest of his body will heal wounds.  Sound reasonable?

vinylspiders replied:
One last thing I am confused about-- let's say you removed a plasmaman's liver. I would expect plasmamen to not take tox damage from that and looking at the code, it seems like that was the intent. However, that is not the case from my testing.

In reagents.dm:

`var/affected_biotype = MOB_ORGANIC` Note that the plasmaman species is not `MOB_ORGANIC`, it is `MOB_MINERAL` and `MOB_HUMANOID`

Then we have this:

```
/datum/reagent/toxin/on_mob_life(mob/living/carbon/affected_mob, delta_time, times_fired)
	if(toxpwr && affected_mob.health > health_required)
		affected_mob.adjustToxLoss(toxpwr * REM * normalise_creation_purity() * delta_time, FALSE, required_biotype = affected_biotype)
		. = TRUE
	..()
```
Note the `required_biotype = affected_biotype` bit.

From /code/modules/mob/living/carbon/damage_procs.dm:
```

/mob/living/carbon/adjustToxLoss(amount, updating_health = TRUE, forced = FALSE, required_biotype)
	if(required_biotype && !(mob_biotypes & required_biotype))
		return

```
Given required_biotype is set to `MOB_ORGANIC` and plasmamen have `MOB_MINERAL `
adjustToxLoss should return here, right? With no tox damage applied? Am I missing something?

vinylspiders replied:
@MrMelbert @timothymtorres After #72664 a thought popped into my head--plasmamen shouldn't be taking organ damage from _any_ toxins that target organics, right? So long as they have their plasmaman liver. And so I came up with a better, more general solution than the one before.

Since we already have affected_biotype why not add a liver_biotype (using the same bitflags as mob_biotype) that matches against that?

By default all of the liver_biotypes are set to MOB_ORGANIC, except plasmamen which are MOB_MINERAL.
This will allow for future additions of species-targeting toxins to not require snowflake code to implement. Now add the relevant biotype and it will just work.

This will make the plasma_liver_metabolism trait go unused--I left it in for now, but can remove it if you'd like. Might be useful somewhere down the line perhaps if you want to specify plasma.

vinylspiders committed:
more refactoring

vinylspiders committed:
Hot Ice is described as 'frozen plasma'

Should also similarly be non-toxic to plasmamen and heal wounds

vinylspiders committed:
Modified handling of plasmaman toxins in liver to a more generalized approach

Better for future proofing if other species or species-targeted chems are added in the future, will make it much simpler and prevent the need for special cases. Just add the bitflag to the liver for the desired liver biotype.

vinylspiders committed:
Merge branch 'master' into plasmaman-tox-fix

vinylspiders committed:
Moves the plasmaman wound healing back into handle_chems, return false to do normal metabolism

vinylspiders committed:
removing extra tab in the diffs

vinylspiders committed:
breathing in plasma from internals now heals when afflicted with plasma fixation virus

vinylspiders committed:
fixed indentation mistake

vinylspiders committed:
fixed a typo in the comments

vinylspiders committed:
Plasma fixation heals liver damage

No other organ damage is healed for now, this is just to prevent an ironic death from injecting yourself with plasma while having the plasma fixation virus.

vinylspiders committed:
Merge branch 'master' into plasmaman-tox-fix

vinylspiders committed:
indentation issue

vinylspiders committed:
Merge branch 'plasmaman-tox-fix' of https://github.com/vinylspiders/tgstation into plasmaman-tox-fix

vinylspiders committed:
Merge branch 'master' into plasmaman-tox-fix

vinylspiders committed:
Merge branch 'plasmaman-tox-fix' of https://github.com/vinylspiders/tgstation into plasmaman-tox-fix

vinylspiders committed:
a better way to handle not taking liver damage from plasma when suffering plasma fixation

And finds a good use for that PLASMA_LOVER_METABOLISM trait.

vinylspiders committed:
Adds a comment

vinylspiders committed:
Replaced numbers with defines and vars for clarity & ease of modification

As well as for debugging purposes.

MrMelbert reviewed (changes requested):
1 thing, rest looks good I think

vinylspiders committed:
Changed PLASMA_LOVER_METABOLISM from a liver trait to a mob trait

To prevent surgical shenanigans.

vinylspiders commented on code/datums/diseases/advance/symptoms/heal.dm:
```diff
@@ -503,24 +521,61 @@
+			ADD_TRAIT(liver, TRAIT_PLASMA_LOVER_METABOLISM, DISEASE_TRAIT)
+		else 
+			REMOVE_TRAIT(liver, TRAIT_PLASMA_LOVER_METABOLISM, DISEASE_TRAIT)
+	return TRUE
```
> Seems like you can get stage 4 plasma fixation, remove your liver, cure the disease, then re-introduce your liver to have permanent plasma lover metab

Aaah I hadn't even considered that! I can see some crafty spessmen figuring that out and abusing it.

In that case, we can probably just forego the organ trait part entirely and have it be a mob trait. Looking at the organ traits system it seems like you'd still run into the same issue of the organ 'remembering' its traits after being removed.

The more I think about it the less an organ trait seems necessary, especially with the new liver biotype stuff. I think you're right about over complicating it here. 

Updated with these changes.

JohnFulpWillard commented on code/modules/surgery/organs/liver.dm:
```diff
@@ -217,6 +220,7 @@
 	name = "reagent processing crystal"
 	icon_state = "liver-p"
 	desc = "A large crystal that is somehow capable of metabolizing chemicals, these are found in plasmamen."
+	liver_biotype = MOB_MINERAL
```
Every single ``affected_biotype`` currently is ``MOB_ORGANIC``, does this do anything?

vinylspiders replied:
Yes. This important because it is what stops plasmamen from taking liver damage from plasma and other organic toxins.

`if(!(liver_biotype & toxin.affected_biotype)) //this particular toxin does not affect this biotype
	continue `

For plasmamen who do not have the MOB_ORGANIC liver biotype this will cause it to skip over all toxins that do not have the MOB_MINERAL affected_biotype flag.

vinylspiders replied:
As for why not use mob biotype here, it's because transplanted and cybernetic livers are a thing. So the liver biotype should be considered independently. All livers are organic except the plasmaman one currently.

More types of livers may be added this way if you want it to be able to process a specific affected_biotype.

vinylspiders committed:
Changed to use already existing var, 'status', adds a new status "ORGAN_INORGANIC"

vinylspiders committed:
Updated comment

vinylspiders committed:
Changed ORGAN_INORGANIC to ORGAN_MINERAL

vinylspiders committed:
Fixes some checks for robotic organs that would wrongly exclude the new ORGAN_MINERAL organ status

timothymtorres commented:
I know you've been working on this for a few weeks and have done a really good job, but I think it might be best to split this PR into smaller ones so maintainers have an easier time to review it since you've got multiple complicated changes happening.  I'm definitely guilty of bloating PRs myself so I know it's annoying when someone asks for this.  I'd split this into like 3 seperate PRs:

- Plasma breathing symptom virus fix
- Plasma and hot ice chems fix for plasmaman chem metabolism
- Mineral organ type for plasma liver that ignores toxins

Do this and maintainers will be able to review stuff faster since the surface area is easier to understand.

timothymtorres commented on code/__DEFINES/mobs.dm:
```diff
@@ -56,6 +56,7 @@
 //Organ defines for carbon mobs
 #define ORGAN_ORGANIC 1
 #define ORGAN_ROBOTIC 2
+#define ORGAN_MINERAL 3 // Used for the plasmaman liver
```
I would refactor all the organ flags to use mob_biotype flags for this.  `MOB_MINERAL`, `MOB_HUMANOID`, etc.  Might be way out of scope for the PR though.

vinylspiders replied:
I agree, and that's something I was wanting to do but it would make this PR even more annoying to review so I withheld for now. I'd have to touch every single piece of code that checks for robotic organs. It is a step in the right direction at least.

timothymtorres commented on code/datums/diseases/advance/symptoms/heal.dm:
```diff
@@ -493,6 +503,14 @@
 		"Transmission 6" = "Increases temperature adjustment rate.",
 		"Stage Speed 7" = "Increases healing speed.",
 	)
+	
+	//vars to easily tweak healing numbers in game for debugging
+	var/healing_per_mol = HEALING_PER_MOL
+	var/healing_per_breath_pressure = HEALING_PER_BREATH_PRESSURE
+	var/max_heal_coefficient_internals = MAX_HEAL_COEFFICIENT_INTERNALS
+	var/max_heal_coefficient_environment = MAX_HEAL_COEFFICIENT_ENVIRONMENT
+	var/max_heal_coefficient_bloodstream = MAX_HEAL_COEFFICIENT_BLOODSTREAM
+	var/base_heal = BASE_HEAL_PLASMA_FIXATION
```
You should remove these and just use the defines directly after you are done debugging.

vinylspiders replied:
alright, done. thanks for looking it over.

timothymtorres commented on code/datums/diseases/advance/symptoms/heal.dm:
```diff
@@ -503,24 +521,59 @@
+	///  the amount of mols in a breath is significantly lower than in the environment so we are just going to use the tank's
+	///  distribution pressure as an abstraction rather than calculate it using the ideal gas equation.
+	///  balanced around a tank set to 4kpa = about 0.2 healing power. maxes out at 0.75 healing power, or 15kpa.
+	if(iscarbon(M))
+		var/mob/living/carbon/C = M
+		var/obj/item/tank/internals/internals_tank = C.internal
```
Don't use single letter vars.

timothymtorres commented on code/datums/diseases/advance/symptoms/heal.dm:
```diff
@@ -503,24 +521,59 @@
 		temp_rate = 4
 
+// We do this to prevent liver damage from injecting plasma when plasma fixation virus reaches stage 4 and beyond
+/datum/symptom/heal/plasma/on_stage_change(datum/disease/advance/A)
+	. = ..()
+	if(!.)
+		return FALSE
+		
+	if(A.stage >= 4)
+		ADD_TRAIT(A.affected_mob, TRAIT_PLASMA_LOVER_METABOLISM, DISEASE_TRAIT)
+	else 
+		REMOVE_TRAIT(A.affected_mob, TRAIT_PLASMA_LOVER_METABOLISM, DISEASE_TRAIT)
+	return TRUE
+
+/datum/symptom/heal/plasma/End(datum/disease/advance/A)
+	. = ..()
+	if(!.)
+		return
+		
+	REMOVE_TRAIT(A.affected_mob, TRAIT_PLASMA_LOVER_METABOLISM, DISEASE_TRAIT)
```
Again, no single letter vars.

vinylspiders replied:
This is how all of the virology code does it. Do you want me to refactor all of it?

Time-Green replied:
We're trying to slowly work away single letter vars. It might seem inconsistent for now, but it's the lesser evil
https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#dont-use-abbreviations

vinylspiders replied:
Gotcha--I will make the change. I'll let someone else worry about refactoring the whole thing.

vinylspiders commented:
> I know you've been working on this for a few weeks and have done a really good job, but I think it might be best to split this PR into smaller ones so maintainers have an easier time to review it since you've got multiple complicated changes happening. I'm definitely guilty of bloating PRs myself so I know it's annoying when someone asks for this. I'd split this into like 3 seperate PRs:
> 
>     * Plasma breathing symptom virus fix
> 
>     * Plasma and hot ice chems fix for plasmaman chem metabolism
> 
>     * Mineral organ type for plasma liver that ignores toxins
> 
> 
> Do this and maintainers will be able to review stuff faster since the LOC surface area is tiny and easy to comprehend. You also gain more GBP since every fix is not stacked into one PR.

ahh the pAiN.

I will hold out hope for at least until after this weekend before doing that. Perhaps I can convince a maintainer to look this over if I talk them through the changes. But I agree in general, that just sounds so unappealing to me at this moment.

vinylspiders committed:
Removal of debugging vars

as per request

vinylspiders committed:
Removes single letter vars

Time-Green commented on code/modules/antagonists/heretic/structures/mawed_crucible.dm:
```diff
@@ -84,7 +84,7 @@
 		var/obj/item/organ/consumed = weapon
-		if(consumed.status != ORGAN_ORGANIC || (consumed.organ_flags & ORGAN_SYNTHETIC))
+		if(consumed.status == ORGAN_ROBOTIC || (consumed.organ_flags & ORGAN_SYNTHETIC))
 			balloon_alert(user, "not organic!")
```
I'll let it slide because no one knows the difference

Time-Green commented on code/modules/surgery/organs/liver.dm:
```diff
@@ -107,6 +107,8 @@
 
 	if(filterToxins && !HAS_TRAIT(liver_owner, TRAIT_TOXINLOVER))
 		for(var/datum/reagent/toxin/toxin in cached_reagents)
+			if(status != toxin.affected_organtype) //this particular toxin does not affect this type of organ
```
Just want to note for other people that it's fine since both can only contain one flag

Time-Green commented:
Sorry for the late merge. The size is fine, sometimes shit just falls through the cracks
